src/librustc_session/config.rs

```diff
@@ -1698,11 +1698,29 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
         }
     }
 
-    if cg.profile_generate.enabled() && cg.profile_use.is_some() {
-        early_error(
-            error_format,
-            "options `-C profile-generate` and `-C profile-use` are exclusive",
-        );
+    if debugging_opts.instrument_coverage && !cg.profile_generate.enabled() {
+        // `-Z instrument-coverage implies `-C profile-generate` (to generate additional LLVM
```

Comment:
It sounds kind of like you're enabling PGO profile generation, which I wouldn't expect. At least I thought the information was separate? @petrhosek would know..

Reply:
@tmandry you marked this comment "resolved". Was that intentional?

Without the `-Cprofile-generate` flag, the LLVM instrprof tooling failed with errors related to various symbols, etc.

I found the missing dependencies were exported with this flag, and I didn't see any significant downside, but it's not unfathomable that I missed something.

Should I follow up? Or did you decide that this isn't a problem?

Reply:
Hmm, I didn't mean to mark it resolved. My concern is that this might enable extra things we don't want (like extra instrumentation passes and extra profiling output when the binary is run).

Reply:
@tmandry You were absolutely right to have these concerns! I spend a day or two trying to debug the coverage instrumentation on my `json5format` crate and its dependencies, and it turns out the problems were because of `-Cprofile-generate`! So I removed the code that enables that flag by default, and added a couple of changes to give the InstrProf libraries what they were looking for, and the problems went away!